docs/book/v5/core-features/error-reporting.md

```diff
@@ -0,0 +1,125 @@
+    - `ErrorReportServiceInterface::class` . `path` must have a value; if the destination file does not exist, it will be created automatically.
+    - `ErrorReportServiceInterface::class` . `tokens` must contain at least one token.
+    - At least one of `ErrorReportServiceInterface::class` . `domain_whitelist`/`ip_whitelist` must have at least one value.
+
+**Note:** The function `checkRequest()` tries to validate the request by checking matches for `domain_whitelist` with `isMatchingDomain()` and for `ip_whitelist` with `isMatchingIpAddress()`.
+If both return `false`, a `ForbiddenException` is thrown and the error message does not get stored.
+
+- **Application-side requirements**:
+    - Send the `Error-Reporting-Token` header with a valid token previously stored in `config/autoload/error-handling.global.php` in the `ErrorReportServiceInterface::class` . `tokens` array.
+    - Send the `Origin` header set to the application's URL; this is the application that send the report.
+
+**Note:**
+
+- The tokens under `ErrorReportServiceInterface::class` . `tokens` do not expire.
+- The log file stores the token value too, making it easy to identify which application sent the error message.
+
+If your post passes all the checks, the message is saved for the developers.
+
+#### Tips and tricks
+
+If there are multiple applications that report errors to your API, you can **assign a different error reporting token** for each.
+The tokens support key-value pairs where
+
+- The **key** is an alias relevant to the assigned application that uses it.
+- The **value** is the token itself.
+
+Example:
+
+```php
+// ...
+return [
+    ...
+    ErrorReportServiceInterface::class => [
+        // ...
+        'tokens' => [
+            'frontend' => '0123456789abcdef0123456789abcdef01234567',
+            'admin' => '9876543210abcdef0123456789abcdef7654321',
+            // other tokens
+        ],
+    ],
+];
+```
+
+The log file will have entries similar to the below:
+
+> [2024-08-29 12:47:00] [0123456789abcdef0123456789abcdef01234567] Demo error message
+
+The inclusion of the token helps you identify the source of the error message.
+In our example, it's the application that uses the `0123456789abcdef0123456789abcdef01234567` token, which is assigned to the application `frontend`.
+
+## How to use it on the Frontend side (Angular example)
+
+The API developer sends a generated token to the frontend developer who will save it in their `environment.staging.ts` and/or `environment.prod.ts`.
+From then on, it's the frontend developer's job to set up an error reporting function similar to the one below.
+
+```javascript
+postError(body: object): Promise<any> {
+     return new Promise((resolve, reject) => {
+      return this.http.post(API_ENDPOINT + 'error-report', body , {headers: new HttpHeaders({'X-Workspace': 'TOKEN', 'Access-Control-Allow-Origin': '*'})})).subscribe({
```

Comment:
```suggestion
      return this.http.post(API_ENDPOINT + 'error-report', body , {headers: new HttpHeaders({'Error-Reporting-Token': 'TOKEN', 'Origin': 'https://example.com'})})).subscribe({
```
The original code snippet does not set the correct headers, I had to fix it because the docs above says:

> - **Application-side requirements**:
>     - Send the `Error-Reporting-Token` header ...
>     - Send the `Origin` header ...